operator/src/main/java/org/bf2/operator/controllers/ManagedKafkaAgentController.java

```diff
@@ -112,6 +112,9 @@ private ManagedKafkaAgentStatus buildStatus(ManagedKafkaAgent resource) {
         } else {
             ConditionUtils.updateConditionStatus(readyCondition, statusValue, null, null);
         }
+        if (!this.observabilityManager.isObservabilityRunning() && readyCondition != null) {
```

Comment:
The status will be non-null here. Removing the check should improve the coverage too :-)
```suggestion
        if (!this.observabilityManager.isObservabilityRunning()) {
```